src/mame/heathkit/tlb.cpp

```diff
@@ -1390,6 +1417,269 @@ ioport_constructor heath_gp19_tlb_device::device_input_ports() const
+ *
+ * New interrupt source
+ *   - horizontal retrace interrupt
+ *
+ */
+heath_imaginator_tlb_device::heath_imaginator_tlb_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	heath_tlb_device(mconfig, HEATH_IMAGINATOR, tag, owner, clock),
+	m_mem_view(*this, "memmap"),
+	m_p_graphic_ram(*this, "graphicram")
+{
+}
+
+void heath_imaginator_tlb_device::device_add_mconfig(machine_config &config)
+{
+	heath_tlb_device::device_add_mconfig(config);
+
+	m_maincpu->set_addrmap(AS_PROGRAM, &heath_imaginator_tlb_device::mem_map);
+	m_maincpu->set_addrmap(AS_IO, &heath_imaginator_tlb_device::io_map);
+	m_maincpu->set_irq_acknowledge_callback(FUNC(heath_imaginator_tlb_device::irq_ack_cb));
+
+	m_crtc->out_hsync_callback().set(FUNC(heath_imaginator_tlb_device::crtc_hsync_w));
+}
+
+void heath_imaginator_tlb_device::device_start()
+{
+	heath_tlb_device::device_start();
+
+	save_item(NAME(m_mem_map));
+	save_item(NAME(m_im2_d1_val));
+	save_item(NAME(m_alphanumeric_mode_active));
+	save_item(NAME(m_graphics_mode_active));
+	save_item(NAME(m_allow_tlb_interrupts));
+	save_item(NAME(m_allow_imaginator_interrupts));
+	save_item(NAME(m_hsync_irq_raised));
+
+	m_maincpu->space(AS_PROGRAM).install_readwrite_tap(0x6000, 0x7fff, "mem_map_update",
+		[this](offs_t offset, u8 &data, u8 mem_mask) { if (!machine().side_effects_disabled()) { tap_6000h(); } },
+		[this](offs_t offset, u8 &data, u8 mem_mask) { if (!machine().side_effects_disabled()) { tap_6000h(); } });
+
+	m_maincpu->space(AS_PROGRAM).install_readwrite_tap(0x8000, 0xbfff, "irq_update",
+		[this](offs_t offset, u8 &data, u8 mem_mask) { if (!machine().side_effects_disabled()) { tap_8000h(); } },
+		[this](offs_t offset, u8 &data, u8 mem_mask) { if (!machine().side_effects_disabled()) { tap_8000h(); } });
+
+	m_alphanumeric_mode_active = true;
+	m_graphics_mode_active = false;
+
+	m_allow_tlb_interrupts = true;
+	m_allow_imaginator_interrupts = false;
+	m_im2_d1_val = 1;
+
+	m_hsync_irq_raised = false;
+
+	m_mem_map = 1;
+
+	m_mem_view.select(m_mem_map);
+}
+
+void heath_imaginator_tlb_device::device_reset()
+{
+	m_mem_map = 1;
+
+	m_mem_view.select(m_mem_map);
+}
+
+/*
+ *  Memory map for the I-100
+ *
+ *    0x0000-0x1fff    Program ROM (Heath TLB board)
+ *    0x2000-0x3fff    Graphics Command Processor(GCP) ROM (I-100)
+ *    0x4000-0x5fff    Scratchpad RAM (Heath TLB board)
+ *    0x6000-0x7fff    Graphics Scratchpad  R/W RAM (I-100)
+ *    0x8000-0xbfff    Graphics Display R/W RAM(I-100)
+ *    0xc000-0xdfff    Optional Graphics Command Processor memory (I-100)
+ *    0xe000-0xffff    Display Memory (Heath TLB board)
+ */
+void heath_imaginator_tlb_device::mem_map(address_map &map)
+{
+	map.unmap_value_high();
+
+	map(0x0000, 0x1fff).view(m_mem_view);
+
+	// H19 standard ROM
+	m_mem_view[0](0x0000, 0x1fff).rom().region("maincpu", 0).unmapw();
+
+	// GCP ROM mapped to 0x0000 on power-up/reset
+	m_mem_view[1](0x0000, 0x1fff).rom().region("maincpu", 0x2000).unmapw();
```

Comment:
You don’t need to put `.unmapw()` if you aren’t trying to unmap something that’s already there.  If your intention is to silence the warning messages about unmapped writes, you can use `.nopw()`.

Also, since the only thing that’s ever mapped here is ROM and it’s never unmapped, you could use `.bankr(...)` rather than a view, and set it up in `device_start` (use a `required_memory_bank` as the object finder for it).